### PR TITLE
meson: Add option to disable translations

### PR DESCRIPTION
**Pull request by daandemeyer**
This speeds up the meson install step by half a second
which, given the trivial changes required to add this
option, makes it worth the effort to support this I think.

Also includes a second commit to disable translations
in mkosi.build.
    
Before:
    
    ‣ Running build script...
    [1/418] Generating version.h with a custom command
    Installing /root/build/po/be.gmo to /root/dest/usr/share/locale/be/LC_MESSAGES/systemd.mo
    Installing /root/build/po/be@latin.gmo to /root/dest/usr/share/locale/be@latin/LC_MESSAGES/systemd.mo
    Installing /root/build/po/bg.gmo to /root/dest/usr/share/locale/bg/LC_MESSAGES/systemd.mo
    Installing /root/build/po/ca.gmo to /root/dest/usr/share/locale/ca/LC_MESSAGES/systemd.mo
    Installing /root/build/po/cs.gmo to /root/dest/usr/share/locale/cs/LC_MESSAGES/systemd.mo
    Installing /root/build/po/da.gmo to /root/dest/usr/share/locale/da/LC_MESSAGES/systemd.mo
    Installing /root/build/po/de.gmo to /root/dest/usr/share/locale/de/LC_MESSAGES/systemd.mo
    Installing /root/build/po/el.gmo to /root/dest/usr/share/locale/el/LC_MESSAGES/systemd.mo
    Installing /root/build/po/es.gmo to /root/dest/usr/share/locale/es/LC_MESSAGES/systemd.mo
    Installing /root/build/po/fr.gmo to /root/dest/usr/share/locale/fr/LC_MESSAGES/systemd.mo
    Installing /root/build/po/gl.gmo to /root/dest/usr/share/locale/gl/LC_MESSAGES/systemd.mo
    Installing /root/build/po/hr.gmo to /root/dest/usr/share/locale/hr/LC_MESSAGES/systemd.mo
    Installing /root/build/po/hu.gmo to /root/dest/usr/share/locale/hu/LC_MESSAGES/systemd.mo
    Installing /root/build/po/id.gmo to /root/dest/usr/share/locale/id/LC_MESSAGES/systemd.mo
    Installing /root/build/po/it.gmo to /root/dest/usr/share/locale/it/LC_MESSAGES/systemd.mo
    Installing /root/build/po/ja.gmo to /root/dest/usr/share/locale/ja/LC_MESSAGES/systemd.mo
    Installing /root/build/po/ko.gmo to /root/dest/usr/share/locale/ko/LC_MESSAGES/systemd.mo
    Installing /root/build/po/lt.gmo to /root/dest/usr/share/locale/lt/LC_MESSAGES/systemd.mo
    Installing /root/build/po/pl.gmo to /root/dest/usr/share/locale/pl/LC_MESSAGES/systemd.mo
    Installing /root/build/po/pt_BR.gmo to /root/dest/usr/share/locale/pt_BR/LC_MESSAGES/systemd.mo
    Installing /root/build/po/ro.gmo to /root/dest/usr/share/locale/ro/LC_MESSAGES/systemd.mo
    Installing /root/build/po/ru.gmo to /root/dest/usr/share/locale/ru/LC_MESSAGES/systemd.mo
    Installing /root/build/po/sk.gmo to /root/dest/usr/share/locale/sk/LC_MESSAGES/systemd.mo
    Installing /root/build/po/sr.gmo to /root/dest/usr/share/locale/sr/LC_MESSAGES/systemd.mo
    Installing /root/build/po/sv.gmo to /root/dest/usr/share/locale/sv/LC_MESSAGES/systemd.mo
    Installing /root/build/po/tr.gmo to /root/dest/usr/share/locale/tr/LC_MESSAGES/systemd.mo
    Installing /root/build/po/uk.gmo to /root/dest/usr/share/locale/uk/LC_MESSAGES/systemd.mo
    Installing /root/build/po/zh_CN.gmo to /root/dest/usr/share/locale/zh_CN/LC_MESSAGES/systemd.mo
    Installing /root/build/po/zh_TW.gmo to /root/dest/usr/share/locale/zh_TW/LC_MESSAGES/systemd.mo
    Installing /root/build/po/pa.gmo to /root/dest/usr/share/locale/pa/LC_MESSAGES/systemd.mo
    
    real    0m1.467s
    user    0m1.064s
    sys     0m0.392s
    
After (with translations disabled):
    
    ‣ Running build script...
    [1/418] Generating version.h with a custom command
    
    real    0m0.925s
    user    0m0.622s
    sys     0m0.301s

### CODE REVIEW

**Review by bluca — APPROVED** [review]
lgtm

**Comment by DaanDeMeyer** [comment]
How do I figure out what specific test failed in autopkgtest?

**Comment by bluca** [comment]
it says it at the end, in this case it's unrelated:

```
FAILED TESTS:  TEST-50-DISSECT
autopkgtest [21:09:42]: test upstream: -----------------------]
autopkgtest [21:09:42]: test upstream:  - - - - - - - - - - results - - - - - - - - - -
upstream             FAIL non-zero exit status 1
autopkgtest [21:09:42]: test boot-smoke: preparing testbed
autopkgtest [21:11:00]: @@@@@@@@@@@@@@@@@@@@ test bed setup
```

File: meson_options.txt [hunk 1]
```diff
@@ -161,6 +161,9 @@ option('man', type : 'combo', choices : ['auto', 'true', 'false'],
 option('html', type : 'combo', choices : ['auto', 'true', 'false'],
        value : 'false',
        description : 'build and install html pages')
+option('translations', type : 'combo', choices : ['auto', 'true', 'false'],
```
Comment (yuwata) [inline meson_options.txt]: Why this is `combo`? It looks `boolean` is enough for this setting.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: meson_options.txt (hunk 1 above)
Comment (daandemeyer) [reply meson_options.txt]: I went with the same approach used by the html option which is also a combo. If a boolean is preferred I'll modify this to be a boolean.

Comment (daandemeyer) [reply meson_options.txt]: Switched to a boolean option.

**Comment by yuwata** [comment]
Nah, you introduced the new setting with `combo` in the first commit, and then changed to `boolean` in the second commit. Please separate the commit properly...

**Comment by DaanDeMeyer** [comment]
Apologies, I forgot about the second commit and just used `amend`. Pushed a new update that modifies the first commit instead of the second one.

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by bluca** [comment]
I can't see the logs from "Unit tests / build (CLANG_ASAN_UBSAN) (pull_request)"  ?

**Comment by DaanDeMeyer** [comment]
Me neither, probably some random failure from Github Actions itself.

**Pull request merged**